pr47-core/src/data/exception.rs

```diff
@@ -1,13 +1,23 @@
 use std::error::Error;
 
 use crate::data::Value;
+use crate::data::wrapper::DynBase;
 
 pub enum UncheckedException {
     ArgCountMismatch { func_ptr: usize, expected: usize, got: usize },
     InvalidBinaryOp { bin_op: char, lhs: Value, rhs: Value, reason: String }
 }
 
+#[cfg(feature = "async")]
+pub type UserUncheckedException = Box<dyn Error + Send + 'static>;
+
+#[cfg(not(feature = "async"))]
+pub type UserUncheckedException = Box<dyn Error + 'static>;
+
+pub type CheckedException = Box<dyn DynBase>;
```

Comment:
Here comes the question. A `CheckedException` may be created by `raise` or `throw` in Pr47 codes. That exceptions will be in the form of  `Value`s, managed by VM heap. Box<dyn DynBase> might not be adequate,

Reply:
Then we need to rewrite our current `Exception` type alot, together with our procedural macro designs.